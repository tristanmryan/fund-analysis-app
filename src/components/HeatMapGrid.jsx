import React from 'react';

const gapColor = gap => {
  if (gap >= 10) return 'bg-green-500 text-white';
  if (gap >= 5) return 'bg-green-300';
  if (gap >= 0) return 'bg-slate-200';
  return 'bg-red-300';
};

const HeatMapGrid = ({ rows = [], onSelect }) => {
  if (!rows.length) return null;
  return (
    <div data-testid="heatmap">
      <div className="grid grid-cols-4 font-semibold border-b text-sm" role="row">
        <div className="p-2">Asset Class</div>
        <div className="p-2 text-right">Benchmark Score</div>
        <div className="p-2 text-right">Median Peer</div>
        <div className="p-2 text-right">Gap</div>
      </div>
      {rows.map(row => (
        <div
          key={row.assetClass}
          role="row"
          className="grid grid-cols-4 border-b last:border-b-0 cursor-pointer hover:bg-gray-50"
          onClick={() => onSelect && onSelect(row.assetClass)}
        >
          <div className="p-2">{row.assetClass}</div>
          <div className="p-2 text-right">{row.benchmarkScore.toFixed(1)}</div>
          <div className="p-2 text-right">{row.medianPeerScore.toFixed(1)}</div>
          <div
            className={`p-2 text-right ${gapColor(row.gap)}`}
            title={`${row.benchmarkScore.toFixed(1)} vs ${row.medianPeerScore.toFixed(1)}`}
          >
            {row.gap >= 0 ? '+' : ''}{row.gap.toFixed(1)}
          </div>
        </div>
      ))}
    </div>
  );
};

export default HeatMapGrid;
