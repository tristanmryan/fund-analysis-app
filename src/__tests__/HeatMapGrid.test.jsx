import { render } from '@testing-library/react';
import HeatMapGrid from '@/components/HeatMapGrid.jsx';

const rows = [
  { assetClass: 'Large Cap', benchmarkScore: 60, medianPeerScore: 50, gap: 10 }
];

test('renders heatmap snapshot', () => {
  const { asFragment } = render(<HeatMapGrid rows={rows} onSelect={() => {}} />);
  expect(asFragment()).toMatchSnapshot();
});
