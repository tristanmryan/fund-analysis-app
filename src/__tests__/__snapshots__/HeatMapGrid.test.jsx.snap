// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders heatmap snapshot 1`] = `
<DocumentFragment>
  <div
    data-testid="heatmap"
  >
    <div
      class="grid grid-cols-4 font-semibold border-b text-sm"
      role="row"
    >
      <div
        class="p-2"
      >
        Asset Class
      </div>
      <div
        class="p-2 text-right"
      >
        Benchmark Score
      </div>
      <div
        class="p-2 text-right"
      >
        Median Peer
      </div>
      <div
        class="p-2 text-right"
      >
        Gap
      </div>
    </div>
    <div
      class="grid grid-cols-4 border-b last:border-b-0 cursor-pointer hover:bg-gray-50"
      role="row"
    >
      <div
        class="p-2"
      >
        Large Cap
      </div>
      <div
        class="p-2 text-right"
      >
        60.0
      </div>
      <div
        class="p-2 text-right"
      >
        50.0
      </div>
      <div
        class="p-2 text-right bg-green-500 text-white"
        title="60.0 vs 50.0"
      >
        +10.0
      </div>
    </div>
  </div>
</DocumentFragment>
`;
